.github/workflows/git-artifacts.yml

```diff
@@ -0,0 +1,313 @@
+name: git-artifacts
+
+on:
+  workflow_dispatch:
+    inputs:
+      build_only:
+        description: 'Optionally restrict what artifacts to build (portable, installer, etc.). Separate artifacts with spaces'
+        required: false
+      ref:
+        description: 'Optionally override which branch to build'
+        required: false
+        default: main
+      repository:
+        description: 'Optionally override from where to fetch the specified ref'
+        required: false
+        default: git-for-windows/git
+      architecture:
+        type: choice
+        description: 'Architecture to build'
+        required: true
+        options: 
+        - x86_64
+        - i686
+
+env:
+  GPG_OPTIONS: "--batch --yes --no-tty --list-options no-show-photos --verify-options no-show-photos --pinentry-mode loopback"
+  HOME: "${{github.workspace}}\\home"
+  USERPROFILE: "${{github.workspace}}\\home"
+  BUILD_ONLY: "${{github.event.inputs.build_only}}"
+  REPOSITORY: "${{github.event.inputs.repository}}"
+  REF: "${{github.event.inputs.ref}}"
+  ARCHITECTURE: "${{github.event.inputs.architecture}}"
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  bundle-artifacts:
+    runs-on: windows-latest
+    outputs:
+      msystem: ${{steps.configure-environment.outputs.MSYSTEM}}
+      mingw_prefix: ${{steps.configure-environment.outputs.MINGW_PREFIX}}
+      bitness: ${{steps.configure-environment.outputs.BITNESS}}
+      git_version: ${{steps.generate-bundle-artifacts.outputs.GIT_VERSION}}
+    steps:
+      - name: Configure environment
+        id: configure-environment
+        run: |
+          case "$ARCHITECTURE" in
+            x86_64)
+              MSYSTEM=MINGW64
+              MINGW_PREFIX="/mingw64"
+              BITNESS=64
+              ;;
+            i686)
+              MSYSTEM=MINGW32
+              MINGW_PREFIX="/mingw32"
+              BTNESS=32
+              ;;
+            *)
+              echo "Unhandled architecture: $ARCHITECTURE"
+              exit 1
+              ;;
+          esac
+          echo "MSYSTEM=$MSYSTEM" >> $GITHUB_ENV
+          echo "MSYSTEM=$MSYSTEM" >> $GITHUB_OUTPUT
+          echo "MINGW_PREFIX=$MINGW_PREFIX" >> $GITHUB_ENV
+          echo "MINGW_PREFIX=$MINGW_PREFIX" >> $GITHUB_OUTPUT
+          echo "BITNESS=$BITNESS" >> $GITHUB_OUTPUT
```

Comment:
The biggest trick with arm64 will be to avoid creating a new tag, and instead re-using the one from the corresponding i686/x86_64 build...

Reply:
Maybe we should use the opportunity to switch things up a bit more and decouple the build from creating the tag? Dunno.

An alternative would be to do what `microsoft/git` does, which creates releases triggered by tags that are pushed to the repository.

In Git for Windows, it is done differently, for two reasons:
- people might fetch the tag prematurely, i.e. before the build succeeded, which means that the build might fail and require more work and the tag would then have to be updated. That can happen e.g. when anything on [the "pre-flight checklist"](https://github.com/git-for-windows/build-extra/blob/main/installer/checklist.txt) fails. It did happen in the past.
- In Git for Windows, we have to deal with embargoed releases (like v2.39.1, last week), where we have to create the artifacts a couple of days in advance, including the tag, and nothing about that must leak to the public before the actual release is published.

So maybe a better alternative would be a separate GitHub workflow that merely creates the tag and the release notes (essentially the `bundle-artifacts` of, say, [the v2.39.0(2) build](https://dev.azure.com/git-for-windows/git/_build/results?buildId=111287&view=artifacts&pathAsName=false&type=publishedArtifacts)).

The `git-artifacts` workflow would then require a reference to said build so that it can download the artifact and use it.

And of course we will eventually also need to port [the Azure Release Pipeline](https://dev.azure.com/git-for-windows/git/_release?_a=releases&view=mine&definitionId=1) that uploads/publishes the artifacts and sends the announcement email. But we can deal with that when the first two workflows are in place.

Reply:
@dscho I've now split up the workflow into `bundle-artifacts.yml` and `git-artifacts.yml`. If no workflow run id of `bundle-artifacts.yml` is provided, it simply takes the latest available one. WDYT?

- Example run of `bundle-artifacts.yml`: https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4036805037
- Example run of `git-artifacts.yml` without `bundle_artifacts_workflow_run_id`: https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4037281758
- Example run of `git-artifacts.yml` with `bundle_artifacts_workflow_run_id` of `4036805037`: https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4037281503

Reply:
@dscho thanks to your changes [here](https://github.com/git-for-windows/git-sdk-32/pull/22) and [here](https://github.com/git-for-windows/build-extra/pull/473), the `i686` run is now progressing a lot further: https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4047585094/jobs/6961778017

However, it's still failing because it was invoked with `--build-src-pkg` [which sets `MINGW_ARCH=mingw64`](https://github.com/git-for-windows/build-extra/blob/9544308613847e67a6863e32d080d701581b5a3a/please.sh#L3695-L3698). That obviously doesn't work in the `git-sdk-32`:

```
==> Finished making: mingw-w64-git 2.39.1.windows.1.529.g10a93f8bfd-1 (Mon Jan 30 20:07:31 2023)
=> WARNING: You don't have the required toolchain installed for mingw64.
=> WARNING: To install it run: 'pacman -S mingw-w64-x86_64-toolchain'
==> Making package: mingw-w64-git 2.39.1.windows.1.529.g10a93f8bfd-1 (Mon Jan 30 20:13:04 2023)
==> Retrieving sources...
/usr/bin/git: line 3: /mingw64/bin/git.exe: No such file or directory
==> ERROR: /usr/src/MINGW-packages/mingw-w64-git/git is not a clone of https://github.com/git-for-windows/git.git
    Aborting...
Could not build mingw-w64-git
```

What's your preferred approach for this? Only build the source package on `x86_64`? Happy to provide a PR but would first like to know which way to go. Thanks!

Reply:
> What's your preferred approach for this? Only build the source package on `x86_64`?

I think that's the best course of action, yes.